### PR TITLE
Model binding with generic types /7

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes https://github.com/dotnet/AspNetCore.Docs/issues/29571

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/mvc/models/model-binding.md](https://github.com/dotnet/AspNetCore.Docs/blob/5b2c49632ba8c7f7e743ab1db90bedcd994a4f4e/aspnetcore/mvc/models/model-binding.md) | [Model Binding in ASP.NET Core](https://review.learn.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?branch=pr-en-us-29573) |

### CODE REVIEW

File: aspnetcore/mvc/models/model-binding.md [hunk 1]
```diff
@@ -353,7 +353,7 @@ The `[Bind]` attribute can be used to protect against overposting in *create* sc
 ```csharp
 [HttpPost]
 public IActionResult OnPost(
-    [ModelBinder(typeof(MyInstructorModelBinder))] Instructor instructor)
+    [ModelBinder<MyInstructorModelBinder>] Instructor instructor)
```
Comment (captainsafia) [inline aspnetcore/mvc/models/model-binding.md:356]: Do we need to add a guard here so we only render the generic attribute for docs targeting 8.0 and above?

File: aspnetcore/mvc/controllers/filters/samples/8.x/FiltersSample/Controllers/ExceptionController.cs [hunk 2]
```diff
@@ -1,13 +1,13 @@
-﻿using FiltersSample.Filters;
+using FiltersSample.Filters;
 using Microsoft.AspNetCore.Mvc;
 
 namespace FiltersSample.Controllers;
 
 // <snippet_Class>
-[TypeFilter(typeof(SampleExceptionFilter))]
+[TypeFilter<SampleExceptionFilter>]
```
Comment (captainsafia) [inline aspnetcore/mvc/controllers/filters/samples/8.x/FiltersSample/Controllers/ExceptionController.cs:7]: I'm assuming that based on the directory these will only surface for 8.x and above docs?

**Review by captainsafia — APPROVED** [review]
Looks good! Some questions inline to confirm that we aren't rendering this for docs pages that don't need it.

File: aspnetcore/mvc/controllers/filters/samples/8.x/FiltersSample/Controllers/ExceptionController.cs (hunk 2 above)
Comment (Rick-Anderson) [reply aspnetcore/mvc/controllers/filters/samples/8.x/FiltersSample/Controllers/ExceptionController.cs:7]: > 

Yes, these changes only show up in 8.0 and later.

**Pull request merged**